### PR TITLE
Workflow generalization

### PR DESCRIPTION
**Pull request by SebastianStehle**

### CODE REVIEW

File: src/Squidex.Domain.Apps.Core.Model/Contents/StatusChange.cs [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-//  Copyright (c) Squidex UG (haftungsbeschraenkt)
+//  Copyright (c) Squidex UG (haftungsbeschränkt)
 //  All rights reserved. Licensed under the MIT license.
 // ==========================================================================
 
 namespace Squidex.Domain.Apps.Core.Contents
 {
     public enum StatusChange
     {
-        Archived,
```
Comment (sauravvijay) [inline src/Squidex.Domain.Apps.Core.Model/Contents/StatusChange.cs]: why have we removed Archived status?

Comment (SebastianStehle) [reply src/Squidex.Domain.Apps.Core.Model/Contents/StatusChange.cs]: It is not the status. It is the transition from Draft => Archived or from Published to Archived.

But when we generalize the workflow we cannot assume anymore that there is a new Archived state, so I have removed this. Published is a special case, it must always exist.

File: tools/Migrate_01/OldTriggers/ContentChangedTriggerSchema.cs [hunk 2]
```diff
@@ -50,19 +50,9 @@ public ContentChangedTriggerSchemaV2 Migrate()
...
-
             if (SendArchived)
             {
-                conditions.Add($"event.type == '{EnrichedContentEventType.Archived}'");
-            }
-
-            if (SendRestore)
-            {
-                conditions.Add($"event.type == '{EnrichedContentEventType.Restored}'");
+                conditions.Add($"event.status == 'Archived'");
```
Comment (sauravvijay) [inline tools/Migrate_01/OldTriggers/ContentChangedTriggerSchema.cs]: Is this correct?

Comment (SebastianStehle) [reply tools/Migrate_01/OldTriggers/ContentChangedTriggerSchema.cs]: No, should be 'Draft', thx

Comment (SebastianStehle) [reply tools/Migrate_01/OldTriggers/ContentChangedTriggerSchema.cs]: No,  everything is fine. The Change here looks just a little bit strange.

File: tests/Squidex.Domain.Apps.Entities.Tests/Contents/ContentGrainTests.cs [hunk 3]
```diff
@@ -337,7 +365,7 @@ public async Task ChangeStatus_should_create_events_and_update_state_when_archiv
 
             LastEvents
                 .ShouldHaveSameEvents(
-                    CreateContentEvent(new ContentStatusChanged { Status = Status.Archived, Change = StatusChange.Archived })
+                    CreateContentEvent(new ContentStatusChanged { Status = Status.Archived })
```
Comment (Ben-Fletcher-UK) [inline tests/Squidex.Domain.Apps.Entities.Tests/Contents/ContentGrainTests.cs]: Has Archived been removed as a possible state now? I have noticed it's gone from the ContentChangedTriggerHandlerTests.cs but it's still in here.

Comment (SebastianStehle) [reply tests/Squidex.Domain.Apps.Entities.Tests/Contents/ContentGrainTests.cs]: It is part of the default implementation.

**Pull request closed without merging**